### ISSUE TITLE
The acc is small

### ISSUE THREAD

**Issue by gaozhiguang** [neulab/guided_summarization#22]
Hi, thanks for the nice work.
When i run bert, i met the problem that the acc is too small, i follow the readme instructions to process my data, do you know why?
![image](https://user-images.githubusercontent.com/34132216/125552145-a9272ac6-65cb-4ca5-a520-86c3267e4b74.png)

**Comment by zdou0830** [comment]
Hi, to debug, you can first run PreSumm with your data and see if you can reproduce their results. If yes, then you can check again if you run `highligted_sentence_data.py`  correctly.

**Comment by gaozhiguang** [comment]
Hi, when i run the code, i met the version problem, so i change the code like this:
![image](https://user-images.githubusercontent.com/34132216/125553937-d39570b4-e1ac-4de4-89f3-f366f32a99bf.png)
does the ~a == 1-a here?

**Comment by zdou0830** [comment]
Please use `pip install -r requirements.txt` to install the packages as in README.

**Comment by gaozhiguang** [comment]
Hi, the data works well for Presum, and i use the highligted_sentence_data.py to process my data.
And something is strange is that the acc increase first and then suddenly decline.
![image](https://user-images.githubusercontent.com/34132216/125877508-60f80278-589c-49ed-9c6e-8f39320261ce.png)

**Comment by zdou0830** [comment]
Hi, I'm not sure what happened. I would suggest you to 1) make sure your pytorch version is 1.1.0; 2) make sure the training hyper-parameters are exactly the same with https://github.com/neulab/guided_summarization/blob/master/bert/train.sh; 3) try a different random seed.

**Comment by wxdwlai** [comment]
I met the same problem as you, how did you solve it in the end?

**Comment by gaozhiguang** [comment]
> I met the same problem as you, how did you solve it in the end?

Hi, I set the learning rate for decoder from 0.2 to 0.1, and it works fine.

**Comment by bxianghui** [comment]
> > I met the same problem as you, how did you solve it in the end?
> 
> Hi, I set the learning rate for decoder from 0.2 to 0.1, and it works fine.
What is your learning rate of decoder？

**Comment by bxianghui** [comment]
> I met the same problem as you, how did you solve it in the end?
Have you solved it yet

**Comment by JingAnLu** [comment]
So do I,when training after 5650 steps,the acc sharply decrease
![image](https://user-images.githubusercontent.com/50937430/161659702-948c23f0-172e-4b80-8a21-3974dddd3305.png)

**Comment by bxianghui** [comment]
我把数据集换了，然后修改了很多东西，才可以训练。



---Original---
From: ***@***.***&gt;
Date: Tue, Apr 5, 2022 09:24 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [neulab/guided_summarization] The acc is small (#22)




 
So do I,when training after 5650 steps,the acc sharply decrease

**Comment by JingAnLu** [comment]
好兄弟，你改了啥，我看上面有说改decoder的学习率的 😂

**Comment by bxianghui** [comment]
那个可以改，但没效果还是有问题。你就换数据集吧，好像是数据集有问题。



------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年4月5日(星期二) 上午9:35
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [neulab/guided_summarization] The acc is small (#22)





 
好兄弟，你改了啥，我看上面有说改decoder的学习率的 😂

**Comment by JingAnLu** [comment]
太难了 😭

**Comment by bxianghui** [comment]
好兄弟，哪个学校？




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年4月5日(星期二) 上午9:54
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [neulab/guided_summarization] The acc is small (#22)





 
太难了 😭

**Comment by JingAnLu** [comment]
山东某不知名双非

**Comment by bxianghui** [comment]
好好找找其他数据集，改下预处理，应该问题不大




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年4月5日(星期二) 上午10:02
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [neulab/guided_summarization] The acc is small (#22)





 
山东某不知名双非

**Comment by JingAnLu** [comment]
好的，谢谢好兄弟

**Comment by tiexueYL** [comment]
> 好的，谢谢好兄弟

大佬 改了解码器的学习率就好了 你会在bert模型加指导信号吗 求指点，，

**Comment by tiexueYL** [comment]
> 好好找找其他数据集，改下预处理，应该问题不大
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: ***@***.***&gt;; 发送时间: 2022年4月5日(星期二) 上午10:02 收件人: ***@***.***&gt;; 抄送: ***@***.***&gt;; ***@***.***&gt;; 主题: Re: [neulab/guided_summarization] The acc is small (#22) 山东某不知名双非 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: ***@***.***&gt;

大佬 求问怎么在bertabs加指导信号，，求指点，谢谢大佬啊啊

**Comment by bxianghui** [comment]
这个我没做，你要做的话应该先用抽取式先把数据集处理一下，抽出多个不同摘要。然后按照他的数据处理，把英文bert换成中文就行了。&nbsp; 你应该看CNN那个数据集了吧，它就是源文本对应有多个摘要。你要用的话就得搞出来多个摘要。



------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年4月29日(星期五) 上午9:57
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [neulab/guided_summarization] The acc is small (#22)





  
好好找找其他数据集，改下预处理，应该问题不大
 …
 ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: @.&gt;; 发送时间: 2022年4月5日(星期二) 上午10:02 收件人: @.&gt;; 抄送: @.&gt;; @.&gt;; 主题: Re: [neulab/guided_summarization] The acc is small (#22) 山东某不知名双非 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: @.***&gt;
  
大佬 求问怎么在bertabs加指导信号，，求指点，谢谢大佬啊啊

**Comment by tiexueYL** [comment]
谢谢大佬 我用的中文bert和有一个参考摘要的中文数据集 跑的结果还不错  摘要抽取也会 就是不知道怎么加上指导信号 到底是用scripts文件夹的还是highlighted_ sentence.py做 完全一头雾水


| |
YL
|
|
***@***.***
|
签名由网易邮箱大师定制
On 4/29/2022 ***@***.***> wrote：

这个我没做，你要做的话应该先用抽取式先把数据集处理一下，抽出多个不同摘要。然后按照他的数据处理，把英文bert换成中文就行了。&nbsp; 你应该看CNN那个数据集了吧，它就是源文本对应有多个摘要。你要用的话就得搞出来多个摘要。



------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;;
发送时间: 2022年4月29日(星期五) 上午9:57
收件人: ***@***.***&gt;;
抄送: ***@***.***&gt;; ***@***.***&gt;;
主题: Re: [neulab/guided_summarization] The acc is small (#22)






好好找找其他数据集，改下预处理，应该问题不大
…
------------------&nbsp;原始邮件&nbsp;------------------ 发件人: @.&gt;; 发送时间: 2022年4月5日(星期二) 上午10:02 收件人: @.&gt;; 抄送: @.&gt;; @.&gt;; 主题: Re: [neulab/guided_summarization] The acc is small (#22) 山东某不知名双非 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: @.***&gt;

大佬 求问怎么在bertabs加指导信号，，求指点，谢谢大佬啊啊

**Comment by bxianghui** [comment]
如果他的highlighted_ sentence.py是正常的话，在训练前跑一下这个就是指导信息了吧（应该是，我没用，它英文数据集我用了但是根本跑不上去acc，就没再跑了），然后训练



------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年4月29日(星期五) 上午10:43
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [neulab/guided_summarization] The acc is small (#22)





 谢谢大佬 我用的中文bert和有一个参考摘要的中文数据集 跑的结果还不错  摘要抽取也会 就是不知道怎么加上指导信号 到底是用scripts文件夹的还是highlighted_ sentence.py做 完全一头雾水
 
 
 | |
 YL
 |
 |
 ***@***.***
 |
 签名由网易邮箱大师定制
 On 4/29/2022 ***@***.***&gt; wrote：
 
 这个我没做，你要做的话应该先用抽取式先把数据集处理一下，抽出多个不同摘要。然后按照他的数据处理，把英文bert换成中文就行了。&amp;nbsp; 你应该看CNN那个数据集了吧，它就是源文本对应有多个摘要。你要用的话就得搞出来多个摘要。
 
 
 
 ------------------&amp;nbsp;原始邮件&amp;nbsp;------------------
 发件人: ***@***.***&amp;gt;;
 发送时间: 2022年4月29日(星期五) 上午9:57
 收件人: ***@***.***&amp;gt;;
 抄送: ***@***.***&amp;gt;; ***@***.***&amp;gt;;
 主题: Re: [neulab/guided_summarization] The acc is small (#22)
 
 
 
 
 
 
 好好找找其他数据集，改下预处理，应该问题不大
 …
 ------------------&amp;nbsp;原始邮件&amp;nbsp;------------------ 发件人: @.&amp;gt;; 发送时间: 2022年4月5日(星期二) 上午10:02 收件人: @.&amp;gt;; 抄送: @.&amp;gt;; @.&amp;gt;; 主题: Re: [neulab/guided_summarization] The acc is small (#22) 山东某不知名双非 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: @.***&amp;gt;
 
 大佬 求问怎么在bertabs加指导信号，，求指点，谢谢大佬啊啊

**Comment by tiexueYL** [comment]
谢谢大佬 我再整整   当时跑的时候改了解码器学习率 acc正常了
| |
YL
|
|
***@***.***
|
签名由网易邮箱大师定制
On 4/29/2022 ***@***.***> wrote：

如果他的highlighted_ sentence.py是正常的话，在训练前跑一下这个就是指导信息了吧（应该是，我没用，它英文数据集我用了但是根本跑不上去acc，就没再跑了），然后训练



------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;;
发送时间: 2022年4月29日(星期五) 上午10:43
收件人: ***@***.***&gt;;
抄送: ***@***.***&gt;; ***@***.***&gt;;
主题: Re: [neulab/guided_summarization] The acc is small (#22)





谢谢大佬 我用的中文bert和有一个参考摘要的中文数据集 跑的结果还不错 摘要抽取也会 就是不知道怎么加上指导信号 到底是用scripts文件夹的还是highlighted_ sentence.py做 完全一头雾水


| |
YL
|
|
***@***.***
|
签名由网易邮箱大师定制
On 4/29/2022 ***@***.***&gt; wrote：

这个我没做，你要做的话应该先用抽取式先把数据集处理一下，抽出多个不同摘要。然后按照他的数据处理，把英文bert换成中文就行了。&amp;nbsp; 你应该看CNN那个数据集了吧，它就是源文本对应有多个摘要。你要用的话就得搞出来多个摘要。



------------------&amp;nbsp;原始邮件&amp;nbsp;------------------
发件人: ***@***.***&amp;gt;;
发送时间: 2022年4月29日(星期五) 上午9:57
收件人: ***@***.***&amp;gt;;
抄送: ***@***.***&amp;gt;; ***@***.***&amp;gt;;
主题: Re: [neulab/guided_summarization] The acc is small (#22)






好好找找其他数据集，改下预处理，应该问题不大
…
------------------&amp;nbsp;原始邮件&amp;nbsp;------------------ 发件人: @.&amp;gt;; 发送时间: 2022年4月5日(星期二) 上午10:02 收件人: @.&amp;gt;; 抄送: @.&amp;gt;; @.&amp;gt;; 主题: Re: [neulab/guided_summarization] The acc is small (#22) 山东某不知名双非 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: @.***&amp;gt;

大佬 求问怎么在bertabs加指导信号，，求指点，谢谢大佬啊啊

**Comment by MaNing1924382115** [comment]
> 我把数据集换了，然后修改了很多东西，才可以训练。
> […](#)
> ---Original--- From: ***@***.***&gt; Date: Tue, Apr 5, 2022 09:24 AM To: ***@***.***&gt;; Cc: ***@***.******@***.***&gt;; Subject: Re: [neulab/guided_summarization] The acc is small (#22) So do I,when training after 5650 steps,the acc sharply decrease — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: ***@***.***&gt;

**Comment by vedantgannu** [comment]
My accuracy has consistently been bouncing between 5-6 percent even after 4000 steps. I'm using all of original hyperparameters from the z_train.sh file. Am I doing something wrong?
